### PR TITLE
kubectl supports copying files with wild card

### LINKED ISSUE #68222: kubectl cp does not accept wild cards
**FEATURE REQUEST**
/sig cli

As a paranoid admin,
I would like to take a manual backup of all /*.xml files in a folder (think Jenkins)
so that I can preserve a snapshot of the config

e.g. `kubectl cp <namespace>/$jenkins_master_pod:/var/jenkins_home/*.xml .`

**Environment**:
- Kubernetes version:

Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"

Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: AKS (Azure)
- OS (e.g. from /etc/os-release): Mac
- Kernel (e.g. `uname -a`):  Darwin Kernel Version 17.4.0

### PR DESCRIPTION
**Pull request by dixudx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Currently kubectl cp does not accept wild cards, which is quite handy in common usage.

**Which issue(s) this PR fixes**:

Fixes #68222

**Special notes for your reviewer**:
/cc @kubernetes/sig-cli-api-reviews 
/cc soltysh 

**Does this PR introduce a user-facing change?**:

```release-note
kubectl supports copying files with wild card
```

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]
/priority important-longterm

File: pkg/kubectl/cmd/cp/cp.go [hunk 1]
```diff
@@ -338,7 +342,17 @@ func makeTar(srcPath, destPath string, writer io.Writer) error {
 }
 
 func recursiveTar(srcBase, srcFile, destBase, destFile string, tw *tar.Writer) error {
-	filepath := path.Join(srcBase, srcFile)
+	var reg *regexp.Regexp
+	var filepath string
+
+	isRegex := isRegexp(srcFile)
```
Comment (soltysh) [inline pkg/kubectl/cmd/cp/cp.go]: Why you're using a regexp here where https://golang.org/pkg/path/filepath/#Glob exists?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/cp/cp.go (hunk 1 above)
Comment (dixudx) [reply pkg/kubectl/cmd/cp/cp.go]: @soltysh Good point. Updated. PTAL. Thanks.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**